src/Magento/FunctionalTestingFramework/Test/Util/ActionMergeUtil.php

```diff
@@ -106,13 +106,16 @@ private function resolveSecretFieldAccess($resolvedActions)
         $actions = [];
         foreach ($resolvedActions as $resolvedAction) {
             $action = $resolvedAction;
-            $hasSecretRef = $this->actionAttributeContainsSecretRef($resolvedAction->getCustomActionAttributes());
+            $actionHasSecretRef = $this->actionAttributeContainsSecretRef($resolvedAction->getCustomActionAttributes());
```

Comment:
Foreach loop can be changed for faster failing and simpler structure. Consider this workflow:
* Check if secret ref exists
* If secret ref doesn't, continue
* If action not in available actions, exception
* Set new `actionType` based on current action (i.e. `fillField => fillSecretField`)
* Create new action using new `actionType`

Reply:
Changes pushed.